control/control.go

```diff
@@ -603,17 +696,33 @@ func (p *pluginControl) ExpandWildcards(ns core.Namespace) ([]core.Namespace, se
 	return nss, nil
 }
 
-func (p *pluginControl) ValidateDeps(mts []core.Metric, plugins []core.SubscribedPlugin) []serror.SnapError {
+func (p *pluginControl) ValidateDeps(requested []core.RequestedMetric, plugins []core.SubscribedPlugin, configTree *cdata.ConfigDataTree) []serror.SnapError {
 	var serrs []serror.SnapError
-	for _, mt := range mts {
-		errs := p.validateMetricTypeSubscription(mt, mt.Config())
-		if len(errs) > 0 {
-			serrs = append(serrs, errs...)
+	var mts []core.Metric
+	// to prevent errors for control in distributed workflow
+	// where only processors/publishers loaded (empty metric catalog)
+	if len(p.metricCatalog.Keys()) > 0 {
+		for _, r := range requested {
+			nss, err := p.MatchQueryToNamespaces(r.Namespace())
+			if err != nil {
+				// use metric directly from the workflow
+				nss = []core.Namespace{r.Namespace()}
+			}
+
+			for _, ns := range nss {
+				mt := &metric{
+					namespace: ns,
+					version:   r.Version(),
+					config:    configTree.Get(ns.Strings()),
```

Comment:
@IRCody it doesn't make sense, added proper logging and continue with loop

@lmroz added init in case of config tree does not have configuration for current namespace